crates/driver/src/domain/competition/solution/encoding.rs

```diff
@@ -180,98 +181,88 @@ pub fn tx(
-            let aave_lender = addr!("87870Bca3F3fD6335C3F4ce8392D69350B4fA4E2");
-
-            let (flashloan_wrapper, flash_fee_bps) = if flashloan.lender.0 == maker_lender {
-                (&contracts.flashloan_wrappers()[0], eth::U256::zero()) // MAKER
-            } else if flashloan.lender.0 == aave_lender {
-                // TODO: ask AAVE to waive the current 5 BPS fee for our helper contract
-                (&contracts.flashloan_wrappers()[1], eth::U256::from(5)) // AAVE
-            } else {
-                // TODO remove this together with configuration options
-                (&contracts.flashloan_wrappers()[0], eth::U256::zero()) // for driver tests to pass
-            };
-
-            // Allow settlement contract to pull borrowed tokens from flashloan wrapper
-            pre_interactions.insert(
-                0,
-                approve_flashloan(
-                    flashloan.token,
-                    flashloan.amount,
-                    contracts.settlement().address().into(),
-                    flashloan_wrapper,
-                ),
-            );
-
-            // Transfer tokens from flashloan wrapper to user (i.e. borrower) to later allow
-            // settlement contract to pull in all the necessary sell tokens from the user.
-            let tx = contracts::ERC20::at(
-                &contracts.settlement().raw_instance().web3(),
-                flashloan.token.into(),
-            )
-            .transfer_from(
-                flashloan_wrapper.address(),
-                flashloan.borrower.into(),
-                flashloan.amount.0,
-            )
-            .into_inner();
-            pre_interactions.insert(
-                1,
-                eth::Interaction {
-                    target: tx.to.unwrap().into(),
-                    value: eth::U256::zero().into(),
-                    call_data: tx.data.unwrap().0.into(),
-                },
-            );
-
-            // Repayment amount needs to be increased by flash fee
-            let fee_amount = (flashloan.amount.0 * flash_fee_bps).ceil_div(&10_000.into());
-            let repayment_amount = flashloan.amount.0 + fee_amount;
-
-            // Since the order receiver is expected to be the setttlement contract, we need
-            // to transfer tokens from the settlement contract to the flashloan wrapper
-            let tx = contracts::ERC20::at(
-                &contracts.settlement().raw_instance().web3(),
-                flashloan.token.into(),
-            )
-            .transfer_from(
-                contracts.settlement().address(),
-                flashloan_wrapper.address(),
-                repayment_amount,
-            )
-            .into_inner();
-            post_interactions.push(eth::Interaction {
+    {
+        let flashloan_wrapper = contracts
+            .get_flashloan_wrapper(&flashloan.lender)
+            .ok_or(Error::UnknownFlashloanLender(flashloan.lender.0))?;
```

Comment:
Seems like you converted this iterator into a regular loop due to error handling. You can also collect an `Iterator<Item = Result<T>>` into `Result<Vec<T>>` to avoid this huge diff.